jobs/integration/test_opa.py

```diff
@@ -0,0 +1,439 @@
+        control_plane_unit = control_plane_app.units[0]
+        try:
+            await kubectl_apply(
+                templates / "integrator-charm-data" / "vsphere" / "storage-class.yaml",
+                control_plane_unit,
+                tools,
+                model,
+            )
+            yield
+        finally:
+            await kubectl_delete(
+                templates / "integrator-charm-data" / "vsphere" / "storage-class.yaml",
+                control_plane_unit,
+                tools,
+                model,
+            )
+
+    @pytest.fixture(scope="class")
+    async def storage_pool(self, model, k8s_model, tools, storage_class):
+        storage_pool_name = "teststorage"
+        log("Creating storage pool")
+        try:
+            await tools.run(
+                "juju",
+                "create-storage-pool",
+                storage_pool_name,
+                "kubernetes",
+                "-m",
+                tools.k8s_connection,
+                "storage-class=mystorage",
+            )
+            yield storage_pool_name
+        finally:
+            log("Removing storage pool")
+            await tools.run(
+                "juju",
+                "remove-storage-pool",
+                "-m",
+                tools.k8s_connection,
+                storage_pool_name,
+            )
+
+    @pytest.fixture(scope="class")
+    async def update_status_interval(self, tools, k8s_model):
+        interval = await tools.run(
+            "juju",
+            "model-config",
+            "-m",
+            tools.k8s_connection,
+            "update-status-hook-interval",
+        )
+        interval = interval[0].strip()
+        try:
+            log("Decrease update status hook interval")
+            await tools.run(
+                "juju",
+                "model-config",
+                "-m",
+                tools.k8s_connection,
+                "update-status-hook-interval=5s",
+            )
+            yield
+        finally:
+            await tools.run(
+                "juju",
+                "model-config",
+                "-m",
+                tools.k8s_connection,
+                f"update-status-hook-interval={interval}",
+            )
+
+    @pytest.fixture(scope="class")
+    async def opa_controller_manager(self, model, k8s_model, tools):
+        try:
+            log("Deploying the webhook charm")
+            webhook = await k8s_model.deploy(
+                "gatekeeper-controller-manager",
+                channel=tools.charm_channel,
+                trust=True,
+            )
+            yield webhook
+        finally:
+            log("Deleting the gatekeeper charm")
+            await tools.run(
+                "juju",
+                "remove-application",
+                "-m",
+                tools.k8s_connection,
+                "--force",
+                "gatekeeper-controller-manager",
+            )
```

Comment:
this should work too

```suggestion
            await k8s_model.remove_application(webhook.name)
```